web/server/routers/post.ts

```diff
@@ -109,6 +130,72 @@ export const postRouter = router({
       });
     }
   }),
+  offer: procedure
+    .input(
+      z.object({
+        email: z.string(),
+        postOid: zodOidType,
+      })
+    )
+    .mutation(async ({ input }) => {
+      try {
+        const user = await findUserByEmail(input.email);
+        if (!user) {
+          throw new TRPCError({
+            message: "No user with given email exists.",
+            code: "BAD_REQUEST",
+          });
+        }
+      } catch (e) {
+        throw new TRPCError({
+          message: "An unexpected error occured.",
+          code: "INTERNAL_SERVER_ERROR",
+        });
+      }
+      try {
+        const post = await getPost(input.postOid);
+        const email = fosterTypeEmails[post.type];
+        try {
+          const info = await transporter.sendMail({
+            from: '"Angels Among Us Pet Rescue Placements Platform" <bitsofgood.aau@gmail.com>',
+            to: email,
+            subject: "Someone is ready to foster your dog!",
+            text: "User has signed up to foster dog, a stray dog.",
+          });
+        } catch (e) {
+          try {
+            const info = await transporter.sendMail({
+              from: '"Angels Among Us Pet Rescue Placements Platform" <bitsofgood.aau@gmail.com>',
+              to: email,
+              subject: "Someone is ready to foster your dog!",
+              text: "User has signed up to foster dog, a stray dog.",
+            });
+          } catch (e) {
+            try {
+              const info = await transporter.sendMail({
+                from: '"Angels Among Us Pet Rescue Placements Platform" <bitsofgood.aau@gmail.com>',
+                to: email,
+                subject: "Someone is ready to foster your dog!",
+                text: "User has signed up to foster dog, a stray dog.",
+              });
+            } catch (e) {
+              throw new TRPCError({
+                message: "Unable to send Email.",
+                code: "INTERNAL_SERVER_ERROR",
+              });
+            }
+          }
+        }
+      } catch (e) {
```

Comment:
There's a lot of the same code here - use a loop to track the number of tries instead:
```js
let count = 0;
const maxTries = 3;
while(true) {
    try {
        // email sending logic
        // break out of loop, or return, on success
    } catch (e) {
        if (count++ == maxTries) {
          throw new TRPCError({
                message: "Unable to send Email.",
                code: "INTERNAL_SERVER_ERROR",
              });
        }
    }
}
```